### ISSUE TITLE
Flatpak release

### ISSUE THREAD

**Issue by mbnoimi** [hardinfo2/hardinfo2#13]
Hi,

Have you considered to make a Flatpak release?

**Comment by hwspeedy** [comment]
Hi Muhammad,

Flatpak/snap is not interesting for hardinfo2.

It is used on small SBC and PC's - and the work is done to support alot of distro for the last 10 years.
Binary packages for download are also ready - but this program should continue/reappear in fedora/debian repos - it has a history and deserves the space.

**Comment by JaneSmith** [comment]
I wish this were available as a Flatpak on Flathub. Flatpak is the native package manager for my distro. It's a pain to install software through other means.

**Comment by hwspeedy** [comment]
Hi @JaneSmith 

> Flatpak is the native package manager for my distro.

Okay, I will take the bate - What distro is that?

**Comment by JaneSmith** [comment]
In my case, it's Fedora Silverblue. It's an immutable/atomic distro where Flatpak is the primary means of installing apps, and it's both discouraged and less convenient to install through other means. There are other Flatpak-centric distros besides Fedora Silverblue, such as Endless OS.

**Comment by hwspeedy** [comment]
**Ubuntu Packaging wanted**

Yeah, currently nobody with the skills, likes Ubuntu and hardinfo2 enough to put the backports into all maintained Ubuntu repos. 
Hardinfo2 can already be build as package for all maintained Ubuntu versions along with Mint.

See downloads: https://hardinfo2.org/download

**Comment by mbnoimi** [comment]
> See downloads: https://hardinfo2.org/github?latest_prerelease

![Image](https://github.com/user-attachments/assets/8c282adf-3419-49be-a37a-6ed74c485c5b)

This occurs with Linux Mint 22.1 (Ubuntu 24.04 based)

**Comment by hwspeedy** [comment]
@mbnoimi

Thanx for info about updated Mint 22.1 distro - I have added it to build system.

See downloads: https://hardinfo2.org/download

Filename: hardinfo2_2.2.6-LinuxMint-22.1_amd64.deb

**Comment by alastortenebris** [comment]
> Hi Muhammad,
> 
> Flatpak/snap is not interesting for hardinfo2.
> 
> It is used on small SBC and PC's - and the work is done to support alot of distro for the last 10 years. Binary packages for download are also ready - but this program should continue/reappear in fedora/debian repos - it has a history and deserves the space.

I don't get this line of logic. Programs don't magically vanish from every distribution's repositories if they're put on Flathub. Mozilla, KDE, and GNOME all have officially maintained flatpaks of their apps, but you won't see distros remove Dolphin just because there's a flatpak of it.

And there are immutable distros that either highly discourage or flat out disable package overlaying. Bazzite, for example, has using rpm-ostree as "not recommended, use as a last resort". And then there's SteamOS. Technically you can unlock the read only file system to install an application, but it'll be wiped upon a system update.

I implore you to reconsider.

**Comment by hwspeedy** [comment]
@alastortenebris 

Please don't complain when the program already works in distros - as many programs this is a hobby. 
If a special distro does not have the hardinfo2 program - that's okay. If you really want it to have the program - do the hacking work - the Linux way. The people before you have paved the way to make this possible.

**Some programs are just not for flatpaks at the current state - kernel, kernel close programs, hardinfo2.**

In Linux eco system - quality is better - each program does the best it can - that's what makes Linux great.

### My five cents:
### Educate
E1. Stop this insane bots everywhere to get flatpak/snap/appimage popular. 
E2. Educate users - problems, what's good, what's bad, how does it work, who does the work, who pays, etc. 
E3. Any technology will do fine if it is the best in what it does.

### Best solution for now
B1. Flatpak should decide whether it is container or binary distribution
B2. Flatpak (same for appimage/snap/etc.) is not verified builds - the source code review work is simply not done - hence the containers half ass solution. Lets see what immutable distros bring...
B3. **Linux is fast because it is optimized** - containers are the other way - copying all different library versions -> so any **distribution updated program is currently the best solution**. Problem: They are only updated for so long and for each distro - that is why RHEL & UBUNTU are a success....

### Distributing binaries is important to Linux but the task has yet to be solved.
D1. Settle on a few golden library versions and make them LTS (20+ years) and create a real access to programs instead of containers could be one solution. 
D2. Allow for non FOSS to be distributed - Linux needs money too and it's needed to get to the famous desktop year - sofar only steam for home gamers and some commercial LTS distros for paid closed source apps.
D3. **How about a common package manager with application maintainers reviewing the code - all in one place for the FOSS packages? - would it kill 100+ distros that are much the same? Would it ruin the great ideas and work from all these distros?**

It's not easy - but do some work for Linux and maybe you could change the world, thanx.

/hwspeedy

**Comment by alastortenebris** [comment]
Boy if that isn't every single point I've heard against flatpak. And I love the "do some work for Linux maybe and you could change the world". But hey, you said if I wanted the program, I could do it "the Linux way".
...so I did that.
https://github.com/alastortenebris/hardinfo2-flatpak

<img width="1283" height="832" alt="Image" src="https://github.com/user-attachments/assets/bcacb372-31a1-4cd5-b718-ec532a2f800c" />

**Comment by hwspeedy** [comment]
@alastortenebris 

I love that you try - this is a good way to move forward - try and then identify problems and fix them.
If you can get to a point where all information is correct, it can be released as hardinfo2 by PR.

Please continue your work and have fun.

Keep us up to speed on your progress and what changes you need to get the information correct.

I have reopened the thread for others to see and join your project: https://github.com/alastortenebris/hardinfo2-flatpak

Good start - now lets get it right!

/hwspeedy

**Comment by beedell-roke** [comment]
@hwspeedy, I agree with your arguments against containerised solutions, but I don't believe that you've phrased them well, nor do I advise making such arguments unless you can phrase them well.
